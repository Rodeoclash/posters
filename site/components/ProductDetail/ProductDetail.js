import Homelink from "../../components/UI/Homelink";
import Link from "next/link";
import ProductLink from "../ProductLink";
import VariantAddToCart from "../VariantAddToCart";
import VariantImage from "../VariantImage";
import VariantInCart from "../VariantInCart";

import styles from "./ProductDetail.module.css";

const ProductDetail = ({ product }) => {
  const variant = product.variants[0];
  const { title, description } = JSON.parse(product.description);

  // console.log("=== product description data");
  // console.log(description);

  return (
    <div className={styles.root}>
      <div className={styles.image}>
        <VariantImage variant={variant} />
      </div>
      <div className={styles.about}>
        <h3>{title}</h3>
        <p>{description}</p>
        <div>
          <VariantAddToCart variant={variant} />
          <VariantInCart variant={variant} />
          <p>
            <Homelink />
          </p>
        </div>
      </div>
    </div>
  );
};

export default ProductDetail;
