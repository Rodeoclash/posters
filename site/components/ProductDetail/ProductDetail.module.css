.root {
  display: flex;
}

.image {
  padding-right: 0.5rem;
}

.about {
  padding-left: 0.5rem;
}
