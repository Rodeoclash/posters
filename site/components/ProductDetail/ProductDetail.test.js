import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import ProductDetail from "./ProductDetail";

it("renders correctly", () => {
  const product = {
    title: "title",
    description: "{}",
    variants: {
      edges: [
        {
          node: {},
        },
      ],
    },
  };

  const renderer = new ShallowRenderer();

  renderer.render(<ProductDetail product={product} />);

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
