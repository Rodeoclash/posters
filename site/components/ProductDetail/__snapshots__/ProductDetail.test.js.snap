// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <VariantImage />
  </div>
  <div>
    <h3>
      title
    </h3>
    <p />
    <div>
      <VariantAddToCart />
      <VariantInCart />
      <p>
        <Homelink />
      </p>
    </div>
  </div>
</div>
`;
